### ISSUE TITLE
Chain Sync from RPC provider throws an error

### ISSUE THREAD

**Issue by gurukamath** [ethereum/execution-specs#472]
I tried to sync the chain with the current commit [21c4e14](https://github.com/ethereum/execution-specs/commit/21c4e14230a645006b2090aab588c221c2ed4def) and the script throws an error for block 56302.

```
INFO:ethereum_spec_tools.sync:block 56300 applied (took 0.016568107001148746s)
INFO:ethereum_spec_tools.sync:block 56301 applied (took 0.013835476998792728s)
Traceback (most recent call last):
  File "/home/guruprasad/Documents/execution-specs/venv/bin/ethereum-spec-sync", line 33, in <module>
    sys.exit(load_entry_point('ethereum', 'console_scripts', 'ethereum-spec-sync')())
  File "/home/guruprasad/Documents/execution-specs/src/ethereum_spec_tools/sync.py", line 734, in main
    sync.process_blocks()
  File "/home/guruprasad/Documents/execution-specs/src/ethereum_spec_tools/sync.py", line 693, in process_blocks
    self.module("spec").state_transition(self.chain, block)
  File "/home/guruprasad/Documents/execution-specs/src/ethereum/frontier/spec.py", line 193, in state_transition
    ensure(receipt_root == block.header.receipt_root)
  File "/home/guruprasad/Documents/execution-specs/src/ethereum/utils/ensure.py", line 42, in ensure
    raise exception_class()
ethereum.utils.ensure.EnsureError
```

Way to reproduce.

1. Checkout the above mentioned commit
2. `ethereum-spec-sync --rpc-url=<rpc-provider> --optimized --persist=<sync-data-location>`

I was using Infura in this case. Not sure if that has an impact.

**Comment by petertdavies** [comment]
This is a bug in handling of logs. There is a fix in #473.

**Comment by gurukamath** [comment]
This issue is solved after #473 and the block is syncing now without any error
